pkg/controllers/dynakube/extension/conditions.go

```diff
@@ -0,0 +1,32 @@
+package extension
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func setSecretCreatedSuccessCondition(conditions *[]metav1.Condition) {
+	condition := metav1.Condition{
+		Type:    secretConditionType,
+		Status:  metav1.ConditionTrue,
+		Reason:  secretCreatedReason,
+		Message: secretCreatedMessageSuccess,
+	}
+	_ = meta.SetStatusCondition(conditions, condition)
+}
+
+func setSecretCreatedFailureCondition(conditions *[]metav1.Condition, err error) {
+	condition := metav1.Condition{
+		Type:    secretConditionType,
+		Status:  metav1.ConditionFalse,
+		Reason:  secretCreatedReason,
+		Message: fmt.Sprintf(secretCreatedMessageFailure, err),
```

Comment:
just a nitpick, but i think it's not a good idea to use constant here, and better to put content directly:
```suggestion
		Message: "Error creating extensions secret: " + err.Error(),
```

similar to https://github.com/dynatrace/dynatrace-operator/blob/2a85bd2f4aded19b83f259c00fc6f645b4b6e1cd/pkg/util/conditions/error.go#L16-L28


or may be reuse it? not sure we want very specific condition for this case
cc @StefanHauth  @0sewa0 wdyt?

Reply:
https://github.com/Dynatrace/dynatrace-operator/blob/d03faddf20f699667d441657547e31ecd2cb9746/pkg/util/conditions/secret.go#L13-L43

Reply:
Thanks for the input guys! I refactored it a bit and now I use the util.conditions package, [see here](https://github.com/Dynatrace/dynatrace-operator/pull/3485/commits/f288a82fe537eb2e859f3dca4fd369fafcf96094)